### PR TITLE
OCPNODE-1890: Bump k8s api to v0.29.0

### PR DESCRIPTION
**Pull request by sairameshv**
cc: @soltysh

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,30 +1,30 @@
 module github.com/openshift/client-go
 
-go 1.20
+go 1.21.3
```
Comment (soltysh) [inline go.mod:3]: Same comment as in the api, let's stick with 1.20 for now.

File: go.mod [hunk 2]
```diff
@@ -66,3 +66,5 @@ retract v3.9.0+incompatible
 retract v0.0.1
 
 replace k8s.io/kube-openapi => github.com/openshift/kube-openapi v0.0.0-20230816122517-ffc8f001abb0
```
Comment (soltysh) [inline go.mod:68]: You will need to update our fork https://github.com/openshift/kube-openapi/tree/1.29-transition-patches with a PR similar to https://github.com/openshift/kube-openapi/pull/7

**Review by soltysh — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (sairameshv) [reply go.mod]: Addressed

File: go.mod (hunk 2 above)
Comment (sairameshv) [reply go.mod]: Opened a PR - https://github.com/openshift/kube-openapi/pull/8

**Pull request merged**